### DISCUSSION TITLE
PIV support timeline

### DISCUSSION THREAD

**Discussion by smlx** [solokeys/solo2#88] [Q&A] [answered]
I was super excited to have my Solo V2 arrive today but when I plugged it in and tried to use the PIV applet I got this error:

```
selecting piv applet: command failed: smart card error 6a82: data object or application not found
```

I also see this commit (https://github.com/solokeys/solo2/commit/672f2d8575b25b5a89f106d81d0fb604dc72f517) from a couple of months ago so I guess PIV support being disabled is intentional.

@conorpp could you share why it was disabled? And any idea of a timeline for PIV support on the Solo V2?

**Comment by nickray** [comment, marked as answer]
Yeah the PIV app needs a phase of undivided attention to bring it over the finish line. It was disabled because the current state of it can even lead to browser hangs. First priority right now is fixing issues that are surfacing with the FIDO app, after that comes the PIV app.

**Reply by smlx** [reply]
ok thanks for the info :+1:

**Reply by VinSz2** [reply]
any update or roadmap on this?

**Reply by JackThursby252** [reply]
@VinSz2 This project is dead unfortunately.

**Reply by axelsimon** [reply]
@nickray if that is so, would it possible to release the work that has been done and give it to the community? maybe even give the software projects to a foundation?

**Comment by n0xena** [comment]
welp - unfortunately only read this AFTER I just got my solo2 out of the box ...
ordered it specifically because it was said to share firmware with nitrokey3 and hence support piv - should had read the firmware patchnotes ...
guess 40 bucks down the drain

btw: the sk2 firmware is even worse - it doesn't support reset via fido2-token but had to use yubico auth on my phone via nfc ... such waste

**Comment by s00500** [comment]
Is there any status to this ? I am not fully sure if I am just missing something...

❯ solo2 app piv aid
Error: card signaled error: Ok(NotFound) (6A, 82)

**Reply by n0xena** [reply]
read the changelog: it once was an idea but was never finished hence removed/disabled
solokey2only support fido2 - no pgp/piv
it's dead, jim
